coverage_comment/template.py

```diff
@@ -69,5 +70,11 @@ def read_template_file() -> str:
     return resources.read_text("coverage_comment", "default.md.j2")
 
 
-def pct(val):
-    return f"{val:.0%}"
+def pct(val: decimal.Decimal | float) -> str:
+    if type(val) == decimal.Decimal:
+        val *= decimal.Decimal("100")
+        return (
+            f"{val.quantize(decimal.Decimal('1'), rounding=decimal.ROUND_HALF_DOWN)}%"
```

Comment:
I notice that it was `ROUND_DOWN` and it's now `ROUND_HALF_DOWN`.

I think the only thing I'd like to make sure is that we never display 100% unless we actually have all the lines covered. Is this the case here ? (99.9999999999% should round to 99%, not 100%)

Reply:
Thank you for the review!

I'm not quite sure that I understand this comment. Before, we used `ROUND_DOWN` for creating the badges, but this code here is for the template. The previous implementation
```python
def pct(val):
    return f"{val:.0%}"
```

also rounds up or down, depending on how large the decimal place is. I guess this was not intended then?
Whereby this is a bit strange, because also some of the tests seem to assume that the values in the templates aren't always rounded down. This line [here](https://github.com/py-cov-action/python-coverage-comment-action/blob/3f19483f8228c96f9ed1964ed93359b94e87b5fe/tests/integration/test_main.py#L128) should actually assume 85% coverage when rounding down, if I haven't made a bad miscalculation. So we actually also want to round down in the template and this is just an error?

Reply:
Ah, well, maybe it was wrong before...?

What's your gut feeling on this ?

Reply:
> Ah, well, maybe it was wrong before...?

In fact, I'm pretty sure it must have been wrong before that. When we lost the 100% coverage for this repo, the comment still showed 100% [as new coverage](https://github.com/py-cov-action/python-coverage-comment-action/pull/84#issuecomment-1279857308).


> What's your gut feeling on this ?

I definitely agree that it is not good to display 100% unless there really is 100%. For the badges, the previous solution with rounding down definitely works. But I'm not sure if that's the best solution for the comment as well. Now that we actually have the percentages exactly and no longer use float, would there be anything against displaying all percentages in the comment with two (I think two would be the best middle ground between inaccurate and unnecessarily accurate) decimal places and rounded down? Then one would have an exact indication, also for the comparison, and it would be nevertheless only 100% indicated, if also 100% are there. What are your thoughts on this?

Reply:
4 significant digits + rounding down in the comments, that seems good to me